client-lite/src/exe/docs.cpp

```diff
@@ -31,26 +31,30 @@ class ProcessController
         // Do init work like registering signal control handler
         signal(SIGINT, _SignalHandler);
         signal(SIGTERM, _SignalHandler);
+        signal(SIGHUP, _SignalHandler);
     }
 
-    void WaitForShutdown(const std::function<bool()>& fnIsIdle)
+    void WaitForShutdown(const std::function<void()>& fnRefreshConfigs)
     {
         constexpr auto idleTimeout = 60s;
         while (true)
         {
+            /* For now, idle-shutdown mechanism is not applicable when running as a service.
+            The service will be started on boot and will be restarted automatically on failure.
+            SDK can assume docs is running and thus simplifies code for private preview. */
             if (_shutdownEvent.Wait(idleTimeout))
             {
                 break;
             }
+            if (_refreshEvent.IsSignaled())
```

Comment:
>_refreshEvent [](start = 16, length = 13)

Combining the event signaled check with the idle loop means we still have a delay in reading config values.
Instead, accept fnRefreshConfigs in the constructor and store it as a static member and call it from _SignalHandler directly.

Reply:
I did some additional refactoring that might make the class more flexible if we need to support other signals, but let me know what you think

---
In reply to: [594676272](https://github.com/microsoft/do-client/pull/11#discussion_r594676272) [](ancestors = 594676272)

Reply:
It's strange to have a callback function callback into the caller to achieve its task. It is more straightforward to have ProcessController itself handle the shutdown in this case.
Moreover, you now have two copies of the lambda to handle SIGINT and SIGTERM.

In general, refactoring early to support a future unknown requirement is not always good because you never know whether the change as it is today will apply to that requirement.

---
In reply to: [595355095](https://github.com/microsoft/do-client/pull/11#discussion_r595355095) [](ancestors = 595355095,594676272)

Reply:
True, alright will revert

---
In reply to: [595415616](https://github.com/microsoft/do-client/pull/11#discussion_r595415616) [](ancestors = 595415616,595355095,594676272)